test/BlendSwap.js

```diff
@@ -0,0 +1,261 @@
+                expect(result.to).to.equal(alice)
+                expect(result.amount).to.be.bignumber.equal(amount)
+                expect(result.releaseTime).to.be.bignumber.equal(releaseTime)
+            })
+
+            it('should have INITIALIZED status', async function() {
+                const status = (await ctx.swap.status(dummySwapId)).toNumber()
+                expect(status).to.equal(Status.INITIALIZED)
+            })
+
+            it('should not hold secret hash', async function() {
+                expect(await ctx.swap.hashlocks(dummySwapId)).to.equal(ZERO_BYTES32)
+            })
+
+            it('should not hold secret', async function() {
+                expect(await ctx.swap.secrets(dummySwapId)).to.equal(ZERO_BYTES32)
+            })
+
+            it('should decrease Bob\'s token balance', async function() {
+                const bobBalance = await ctx.blend.balanceOf(bob)
+                const expected = toBN(initialBalance).sub(amount)
+                expect(bobBalance).to.be.bignumber.equal(expected)
+            })
+
+            it('should increase Swap contract\'s token balance', async function() {
+                const lockBalance = await ctx.blend.balanceOf(ctx.swap.address)
+                expect(lockBalance).to.be.bignumber.equal(amount)
+            })
+        })
+
+        describe('allows to unlock funds after timeout', async function() {
+            beforeEach(async function() {
+                await testDeploy(bob)
+                await ctx.blend.approve(ctx.swap.address, amount, {from: bob})
+                await ctx.swap.lock(
+                    dummySwapId,
+                    alice,
+                    amount,
+                    releaseTime,
+                    '0x00',
+                    {from: bob}
+                )
+            })
+
+            it('should refund if claimed before revealing secret', async function() {
+                await time.increaseTo(releaseTime)
+                await ctx.swap.claimRefund(dummySwapId, {from: bob})
+            })
+
+            it('should refund if claimed before revealing secret hash', async function() {
+                await ctx.swap.revealSecretHash(dummySwapId, dummySecretHash, {from: bob})
+                await time.increaseTo(releaseTime)
+                await ctx.swap.claimRefund(dummySwapId, {from: bob})
+            })
+
+            it('should reject refund if claimed after revealing secret', async function() {
+                await ctx.swap.revealSecretHash(dummySwapId, dummySecretHash, {from: bob})
+                await ctx.swap.redeem(dummySwapId, dummySecret, {from: alice})
+                await time.increaseTo(releaseTime)
+                await expectRevert(
+                    ctx.swap.claimRefund(dummySwapId, {from: bob}),
+                    "Wrong status"
+                )
+            })
+        })
+
+        describe('allows to supply secret hash', async function() {
```

Comment:
Do we check here that Alice can not supplement an arbitrary hash to swap committed by Bob (which is in `INITIALIZED` state)?